crates/sparrow-scheduler/src/worker.rs

```diff
@@ -0,0 +1,105 @@
+use crate::{queue::*, Error, TaskRef};
+
+pub trait Scheduler {
+    /// Schedule a task for immediate, local execution.
+    ///
+    /// For local queues, this will schedule it as the next task, potentially
+    /// displacing the other task(s) scheduled at the front.
```

Comment:
Is the reasoning for this (the whole attempt-to-use-the-same-core-for-cached-data) explained elsewhere?

Reply:
Not particularly -- that should be part of the top level docs, which I'll add once this settles a bit. Basically, it would just be a reference to "Morsel driven parallelism".